### PR TITLE
Automatically retry failed system-probe kitchen tests

### PR DESCRIPTION
**Pull request by brycekahle**
### What does this PR do?

- Automatically retries failing system-probe kitchen tests up to 2 times. 
- Remove coloring to output that was only needed when multiple kernels were run in the same job
- Force `gotestsum` to output colored output for easier reading.
- Add `tasks/system_probe.py` to list of files triggering system-probe tests.

### Motivation

Prevent flaky tests from blocking pull requests.

### Additional Notes

Due to a [bug](https://github.com/gotestyourself/gotestsum/issues/341) in `gotestsum`, it will also re-run sub-tests in-between the root test and the failing sub-test.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: test/kitchen/test/integration/system-probe-test/rspec_datadog/system-probe-test_spec.rb [hunk 1]
```diff
@@ -81,6 +83,8 @@ def get_timeout(package)
           "--format", "dots",
           "--junitfile", xmlpath,
           "--jsonfile", "/tmp/pkgjson/#{pkg.gsub("/","-")}.json",
+          "--rerun-fails=2",
```
Comment (hmahmood) [inline test/kitchen/test/integration/system-probe-test/rspec_datadog/system-probe-test_spec.rb:86]: What's the difference between this option and the one below?

**Review by usamasaqib — APPROVED** [review]
LGTM. A few questions, but nothing blocking.

File: .gitlab/functional_test/system_probe.yml [hunk 2]
```diff
@@ -28,6 +28,7 @@
     when: always
     paths:
       - $DD_AGENT_TESTING_DIR/kitchen-junit-*.tar.gz
+      - $DD_AGENT_TESTING_DIR/testjson
```
[the remark this answers is not in the document]
Comment (brycekahle) [reply .gitlab/functional_test/system_probe.yml:31]: I was using it to manually examine the output for verification purposes. Not strictly required, but helpful for debugging.

File: .gitlab/functional_test/system_probe.yml [hunk 3]
```diff
@@ -61,6 +62,7 @@ kitchen_test_system_probe_linux_x64_ec2:
     - .kitchen_ec2_location_us_east_1
     - .kitchen_ec2
   needs: [ "tests_ebpf_x64", "prepare_ebpf_functional_tests_x64", "generate_minimized_btfs_x64", "pull_test_dockers_x64" ]
+  retry: 0
```
[the remark this answers is not in the document]
Comment (brycekahle) [reply .gitlab/functional_test/system_probe.yml:65]: Yes, to override the inherited value of `retry: 1`.

**Review by bkabrda — APPROVED** [review]

**Pull request merged**